python/python/glide/async_commands/server_modules/ft_options/ft_create_options.py

```diff
@@ -0,0 +1,429 @@
+            initial_cap (Optional[int]): Initial vector capacity in the index affecting memory allocation size of the index. Defaults to 1024.
+            m (Optional[int]): Number of maximum allowed outgoing edges for each node in the graph in each layer. Default is 16, maximum is 512.
+            ef_contruction (Optional[int]): Controls the number of vectors examined during index construction. Default value is 200, Maximum value is 4096.
+            ef_runtime (Optional[int]): Controls the number of vectors examined during query operations. Default value is 10, Maximum value is 4096.
+        """
+        super().__init__(dim, distance_metric, type)
+        self.initial_cap = initial_cap
+        self.m = m
+        self.ef_contruction = ef_contruction
+        self.ef_runtime = ef_runtime
+
+    def toArgs(self) -> List[str]:
+        """
+        Get the arguments representing the vector field created with HSNW algorithm.
+
+        Returns:
+            List[str]: A list of HNSW algorithm type vector arguments.
+        """
+        args = super().toArgs()
+        if self.initial_cap:
+            args.extend([FtCreateKeywords.INITIAL_CAP, str(self.initial_cap)])
+        if self.m:
+            args.extend([FtCreateKeywords.M, str(self.m)])
+        if self.ef_contruction:
+            args.extend([FtCreateKeywords.EF_CONSTRUCTION, str(self.ef_contruction)])
+        if self.ef_runtime:
+            args.extend([FtCreateKeywords.EF_RUNTIME, str(self.ef_runtime)])
+        return args
+
+
+class VectorField(Field):
+    """
+    Class for defining vector field in a schema.
+    """
+
+    def __init__(
+        self,
+        name: TEncodable,
+        algorithm: VectorAlgorithm,
+        attributes: VectorFieldAttributes,
+        alias: Optional[str] = None,
+    ):
+        """
+        Initialize a new VectorField instance.
+
+        Args:
+            name (TEncodable): The name of the vector field.
+            algorithm (VectorAlgorithm): The vector indexing algorithm.
+            alias (Optional[str]): An alias for the field.
+            attributes (VectorFieldAttributes): Additional attributes to be passed with the vector field after the algorithm name.
+        """
+        super().__init__(name, FieldType.VECTOR, alias)
+        self.algorithm = algorithm
+        self.attributes = attributes
+
+    def toArgs(self) -> List[TEncodable]:
+        """
+        Get the arguments representing the vector field.
+
+        Returns:
+            List[TEncodable]: A list of vector field arguments.
+        """
+        args = super().toArgs()
+        args.append(self.algorithm.value)
+        if self.attributes:
+            attribute_list = self.attributes.toArgs()
+            args.append(str(len(attribute_list)))
+            args.extend(attribute_list)
+        return args
+
+
+class DataType(Enum):
+    """
+    Options for the type of data for which the index is being created.
+    """
+
+    HASH = 1
```

Comment:
all the other enums use the constant strings instead of numbres, why not HASH and JSON? 

e.g. 
```
class VectorAlgorithm(Enum):
    """
    Algorithm for vector type fields used for vector similarity search.
    """

    HNSW = "HNSW"
    """
    Hierarchical Navigable Small World algorithm.
    """
    FLAT = "FLAT"
    """
    Flat algorithm or the brute force algorithm.
    """
```

Reply:
Updated.